src/android/SecureStorage.java

```diff
@@ -19,24 +35,82 @@
 import org.json.JSONObject;
 import org.json.JSONArray;
 
+import javax.crypto.IllegalBlockSizeException;
+
 public class SecureStorage extends CordovaPlugin {
+    private final ExecutorService threadPool = Executors.newCachedThreadPool();
     private static final String TAG = "SecureStorage";
 
     private static final boolean SUPPORTS_NATIVE_AES = Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP;
     private static final boolean SUPPORTED = Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT;
 
     private static final String MSG_NOT_SUPPORTED = "API 19 (Android 4.4 KitKat) is required. This device is running API " + Build.VERSION.SDK_INT;
     private static final String MSG_DEVICE_NOT_SECURE = "Device is not secure";
+    public static final String MIGRATED_FOR_SECURITY = "_SS_MIGRATED_FOR_SECURITY";
```

Comment:
A name that refers to the change could have been used, such as the JIRA issue ID, the version or 1/2 words that identify the issue.

This is both for traceability purposes and in case we need to do another migration.